pytest_tank_test/tk_fixtures.py

```diff
@@ -0,0 +1,232 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import pytest
+import os
+from tk_toolchain.authentication import get_toolkit_user
+from tk_toolchain.testing import create_unique_name
+
+
+@pytest.fixture(scope="session")
+def tk_test_shotgun():
+    """
+    Getting credentials from TK_TOOLCHAIN
+    """
+    sg = get_toolkit_user().create_sg_connection()
+
+    return sg
+
+
+@pytest.fixture(scope="session")
+def tk_test_current_user(tk_test_shotgun):
+    """
+    Get current user
+
+    :returns: The current user id and name
+    """
+    user = get_toolkit_user()
+    username = tk_test_shotgun.find_one(
+        "HumanUser", [["login", "is", str(user)]], ["name"]
+    )
+
+    return username
+
+
+@pytest.fixture(scope="session")
+def tk_test_create_project(tk_test_shotgun):
+    """
+    Generates a fresh Shotgun Project to use with the UI Automation.
+
+    :returns: Current project name and id
+    """
+    # Create or update the integration_tests local storage with the current test run
+    storage_name = "Toolkit UI Automation"
+    local_storage = tk_test_shotgun.find_one(
+        "LocalStorage", [["code", "is", storage_name]], ["code"]
+    )
+    if local_storage is None:
+        local_storage = tk_test_shotgun.create("LocalStorage", {"code": storage_name})
+    # Always update local storage path
+    local_storage["path"] = os.path.expandvars("${SHOTGUN_CURRENT_REPO_ROOT}")
+    tk_test_shotgun.update(
+        "LocalStorage", local_storage["id"], {"windows_path": local_storage["path"]}
+    )
+
+    # Make sure there is not already an automation project created
+    project_name = create_unique_name("Toolkit UI Automation")
+    filters = [["name", "is", project_name]]
+    existed_project = tk_test_shotgun.find_one("Project", filters)
+    if existed_project is not None:
+        tk_test_shotgun.delete(existed_project["type"], existed_project["id"])
+
+    # Create a new project with the Film VFX Template
+    project_data = {
+        "sg_description": "Project Created by Automation",
+        "name": project_name,
+        "tank_name": project_name,
+    }
+    new_project = tk_test_shotgun.create("Project", project_data)
+
+    return new_project
+
+
+@pytest.fixture(scope="session")
+def tk_test_create_entities(
```

Comment:
Same comment here about the use of `create`.

I think `tk_test_entities` would be a better name for the same reasons as above.